clink/app/src/history/history_db.cpp

```diff
@@ -425,7 +425,7 @@ line_id_impl read_lock::line_iter::next(str_iter& out)
         bool eating_ctag = m_eating_ctag;
 
         for (; start != last; ++start, --m_remaining)
-            if (unsigned(*start) > 0x1f)
+            if (unsigned(*start) > 0x1a)
```

Comment:
Can you please describe an example where `0x1b` should be stored and displayed, and explain why?  I think that's going to cause ANSI escape codes to get handled, which will be different than

Also, why are `0x1c` through `0x1f` allowed?

Or, why isn't it allowing all control codes other than `0x00`, `0x0a`, and `0x0d`?

Reply:
The `history` command from a conhost.exe console window prints the ESC as a literal, but in a ConEmu or Windows Terminal console window the ESC gets interpreted as an ANSI escape code.

There are both pros and cons to this.

I've changed history_db.cpp to only consider `0x00`, `0x0a`, and `0x0d` as command delimiters.
I've changed history.cpp to translate control codes to `^A`, `^[`, etc when printing, unless stdout is redirected (in which case something may be trying to parse the output, and the strings should be untranslated).

Reply:
0ac7a44ae25b58ff71efb6050c3c4de98d219b04 addresses this motivation for the PR, in a little different manner.